### PR TITLE
DE7615 Update In-Stock Status on Inventory Import

### PR DESCRIPTION
**Pull request by kojiromike**

### CODE REVIEW

**Comment by rgabruel** [comment]
LGTM

**Comment by scottvanbrug** [comment]
:+1:

File: src/app/code/community/EbayEnterprise/Eb2cInventory/Test/Model/Feed/Item/InventoriesTest.php [hunk 1]
```diff
@@ -146,128 +146,214 @@ public function testUpdateInventories()
...
+     *
+     * @dataProvider provideProductQuantities
+     */
+    public function testUpdateProductQuantity($oldQty, $newQty)
+    {
+        $isChange = ($oldQty !== $newQty);
+        $stockItem = $this->getModelMock('cataloginventory/stock_item', ['getQty', 'setQty']);
         $stockItem
-            ->expects($this->once())
+            ->expects($this->any())
```
Comment (scottvanbrug) [inline src/app/code/community/EbayEnterprise/Eb2cInventory/Test/Model/Feed/Item/InventoriesTest.php]: `expects($this->any())` is essentially optional as of PHPUnit 4.0 (so long as the object being mocked doesn't have a `method` method). There is also a shortcut method for `will($this->returnValue())`. So you could make lines like these as `$mockObject->method('someMethod')->willReturn($someValue);`. Not a huge difference but a bit less verbose.

File: src/app/code/community/EbayEnterprise/Eb2cInventory/Test/Model/Feed/Item/InventoriesTest.php [hunk 2]
```diff
@@ -146,128 +146,214 @@ public function testUpdateInventories()
...
+        $feedItemInventories = $this->getModelMockBuilder('eb2cinventory/feed_item_inventories')
+            ->setMethods(['updateProductQuantity', 'updateProductInStockStatus'])
+            ->getMock();
+        $feedItemInventories
+            ->expects($this->any())
+            ->method('updateProductQuantity')
+            ->will($this->returnValue($quantityChanged));
+        $feedItemInventories
+            ->expects($this->any())
+            ->method('handleNotManagedStock')
```
Comment (scottvanbrug) [inline src/app/code/community/EbayEnterprise/Eb2cInventory/Test/Model/Feed/Item/InventoriesTest.php]: Think this should be a stub of `updateProductInStockStatus`, not `handleNotManagedStock`.

File: src/app/code/community/EbayEnterprise/Eb2cInventory/Test/Model/Feed/Item/InventoriesTest.php [hunk 3]
```diff
@@ -146,128 +146,214 @@ public function testUpdateInventories()
...
         $fii->updateInventories($sampleFeed); // Just verify the right inner methods are called.
     }
+
     /**
+     * Provide product quantities for testing setting the product quantities.
+     *
+     * @return array
      */
-    public function testSetProdQty()
+    protected function provideProductQuantities()
```
Comment (scottvanbrug) [inline src/app/code/community/EbayEnterprise/Eb2cInventory/Test/Model/Feed/Item/InventoriesTest.php]: This needs to be public to be a data provider method.

**Comment by scottvanbrug** [comment]
:+1:

**Pull request merged**